dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardUsersServiceIT.java

```diff
@@ -0,0 +1,138 @@
+package edu.harvard.iq.dataverse.dashboard;
+
+import com.rometools.utils.Lists;
+import edu.harvard.iq.dataverse.DataverseSession;
+import edu.harvard.iq.dataverse.RoleAssigneeServiceBean;
+import edu.harvard.iq.dataverse.UserServiceBean;
+import edu.harvard.iq.dataverse.arquillian.arquillianexamples.WebappArquillianDeployment;
+import edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean;
+import edu.harvard.iq.dataverse.engine.command.exception.PermissionException;
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUser;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.transaction.api.annotation.TransactionMode;
+import org.jboss.arquillian.transaction.api.annotation.Transactional;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.platform.commons.util.StringUtils;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+
+import javax.ejb.EJB;
+import javax.inject.Inject;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Arquillian.class)
+@Transactional(TransactionMode.ROLLBACK)
+public class DashboardUsersServiceIT extends WebappArquillianDeployment {
+
+    @Inject
+    private DataverseSession dataverseSession;
+    @EJB
+    private AuthenticationServiceBean authenticationServiceBean;
+    @Inject
+    private DashboardUsersService dashboardUsersService;
+    @Inject
+    private RoleAssigneeServiceBean roleAssigneeService;
+    @Inject
+    private UserServiceBean userServiceBean;
+
+    @Before
+    public void setUp() {
+        dataverseSession.setUser(authenticationServiceBean.getAdminUser());
+    }
+
+    @Test
+    public void shouldRevokeAllRolesForUser() {
+        // given
+        AuthenticatedUser userWithRoles = userServiceBean.getAuthenticatedUserList("","",100, 0)
+                .stream()
+                .filter(u -> u.getId().equals(2L))
+                .findFirst()
+                .get();
+
+        // when
+        dashboardUsersService.revokeAllRolesForUser(userWithRoles);
+
+        // then
+        AuthenticatedUser dbUserWithoutRoles = userServiceBean.getAuthenticatedUserList("","",100, 0)
+                .stream()
+                .filter(u -> u.getId().equals(2L))
+                .findFirst()
+                .get();
+
+        assertTrue(StringUtils.isBlank(dbUserWithoutRoles.getRoles()));
+        assertNull(roleAssigneeService.getUserExplicitGroups(dbUserWithoutRoles)
+                .stream()
+                .filter(s -> "1-rootgroup".equals(s))
+                .findAny()
+                .orElse(null));
```

Comment:
Probably you should check if this is empty list

Reply:
Looks like `Collectors.toList()` in `RoleAssigneeServiceBean.getUserExplicitGroups(RoleAssignee ra)` can return `null`. I have specifically made it '.orElse(null)` because there were nulls in object returned and not an empty list.

Reply:
Can you check again? What exactly it returns?
For me assert:
`assertThat(roleAssigneeService.getUserExplicitGroups(dbUser), is(empty()));`
Passes

Reply:
Yeah, looks like i had this option checked
![image](https://user-images.githubusercontent.com/51329176/70450008-43efc580-1aa3-11ea-80f9-2b91b827b42d.png)

So for empty lists I've been seeing nulls.